sdks/java/core/src/main/java/org/apache/beam/sdk/io/CompressedSource.java

```diff
@@ -384,7 +384,7 @@ protected final boolean isSplittable() throws Exception {
    */
   @Override
   protected final FileBasedReader<T> createSingleFileReader(PipelineOptions options) {
-    if (channelFactory instanceof FileNameBasedDecompressingChannelFactory) {
+    if (channelFactory instanceof FileNameBasedDecompressingChannelFactory && isSplittable()) {
```

Comment:
Is the logic here that "if it's not splittable, then create a CompressedReader"? Perhaps it should be called UnsplittableReader then?